### PR TITLE
UPPER Case QR Code can be accepted (EXPOSUREAPP-3546)

### PR DESCRIPTION
**Pull request by AndroidMedaGalaxy**
## Description

QR Codes in Upper cases can now be accpted

### Implmentation info
REGEX adjusted and test cases have been added

Please click [here](https://regex101.com/r/RIDKZC/2/) if you want to test more QR code scnearios

## Steps to reproduce
step 1. Try the [QR Code generator](https://www.qr-code-generator.com/) and use the following examples.
HTTPS://LOCALHOST/?123456-12345678-1234-4dA7-b166-B86d85475064
https://localhost/?123456-12345678-1234-4DA7-B166-B86D85475ABC

step 2. modify them however you like (Upper case or lowercase letter) 
step 3. Scan them with the phone.
Step 4. You should not see the error for invalid QR

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/service/submission/ScanResultTest.kt [hunk 1]
```diff
@@ -49,5 +74,7 @@ class ScanResultTest {
     @Test
     fun extractGUID() {
         QRScanResult("https://localhost/?$guid").guid shouldBe guid
+        QRScanResult("https://localhost/?$lowerCaseGuid").guid shouldBe lowerCaseGuid
+        QRScanResult("https://localhost/?$mixedCaseGuid").guid shouldBe mixedCaseGuid
```
Comment (mlenkeit) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/service/submission/ScanResultTest.kt:78]:
```suggestion
        QRScanResult("https://localhost/?$mixedCaseGuid").guid shouldBe mixedCaseGuid
        QRScanResult("HTTPS://LOCALHOST/?$guid").guid shouldBe guid
        QRScanResult("HTTPS://LOCALHOST/?$lowerCaseGuid").guid shouldBe lowerCaseGuid
        QRScanResult("HTTPS://LOCALHOST/?$mixedCaseGuid").guid shouldBe mixedCaseGuid
```

Comment (mlenkeit) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/service/submission/ScanResultTest.kt:78]: Let's make sure that upper/lower case protocal and hostname don't change the guid.

Comment (ritsam) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/service/submission/ScanResultTest.kt:78]: Test cases have been added

**Review by mlenkeit — APPROVED** [review]
LGTM 👍

**Review by harambasicluka — CHANGES_REQUESTED** [review]
This feature is for 1.6.0. Please adjust it accordingly, I changed the base branch to release/1.6.x.

**Comment by mlenkeit** [comment]
I think this PR needs to be rebased after Luka changed the base branch...

**Comment by harambasicluka** [comment]
Small remark, nothing to change now: the ticket is a bug, according to our internal wiki that means the branch has to start with `fix/` and also the ticket number should be included: `fix/3546-qr-scanner-upper-case-guid`.

**Comment by harambasicluka** [comment]
Close this as there where some problems with the base branch, new PR can be found here #1524 .

**Pull request closed without merging**